app/content/team/users/[user]/index.js

```diff
@@ -1,23 +1,129 @@
+import { useQueryClient } from "@tanstack/react-query";
 import { useLocalSearchParams, useNavigation } from "expo-router";
+import {
+  collection,
+  deleteDoc,
+  deleteField,
+  doc,
+  getDocs,
+  query,
+  setDoc,
+  updateDoc,
+  where,
+} from "firebase/firestore";
+import { useState } from "react";
 import { StyleSheet, View } from "react-native";
-import { Appbar } from "react-native-paper";
+import { Appbar, Divider, Menu } from "react-native-paper";
 import { SafeAreaView } from "react-native-safe-area-context";
 import { themeColors } from "~/Constants";
+import DialogComponent from "~/components/dialog";
 import DrillList from "~/components/drillList";
 import EmptyScreen from "~/components/emptyScreen";
 import ErrorComponent from "~/components/errorComponent";
 import Header from "~/components/header";
 import Loading from "~/components/loading";
 import PaperWrapper from "~/components/paperWrapper";
 import ProfileCard from "~/components/profileCard";
+import { currentAuthContext } from "~/context/Auth";
+import { db } from "~/firebaseConfig";
 import { useBestAttempts } from "~/hooks/useBestAttempts";
 import { useDrillInfo } from "~/hooks/useDrillInfo";
 import { useEmailInfo } from "~/hooks/useEmailInfo";
 import { useUserInfo } from "~/hooks/useUserInfo";
 
+//A function to add a user to the blacklist table with a timestamp
+async function blacklistUser(userId, userData) {
+  //Create new document with userId as the id and a time field
+  await setDoc(doc(db, "teams", "1", "blacklist", userId), {
+    time: Date.now(),
+    name: userData["name"],
+  });
+
+  //remove users data
+  removeUser(userId);
+}
+
+async function removeUser(userId) {
+  //Remove all attempts from attempts table with UID == userID
```

Comment:
use Firebase transaction, and put these functions in the hooks folder instead of here so you don't have to import query and stuff

Reply:
Wait why should these functions be in the hooks folder? Do you just want them somewhere so they can used elsewhere? Would they be considered hooks? should we make a lib directory for functions like this?

Reply:
they really shouldn't, but since they use Firebase functions anyways and are similar, I put them in the hooks function. I wanted them to be hooks but we need these functions in if statements so hooks won't work.